### PR TITLE
Add note to deployment docs about handling new versions.

### LINKED ISSUE #20369: Add documentation detailing how we automatically load new versions of your application
See this twitter thread:
https://twitter.com/timer150/status/1341044363573473282

It seems we're currently missing docs covering that we automatically load new versions of your application, and that there's no special or custom code necessary.

We should explain that `<Link>` temporarily works like a normal `<a>` tag to load the new version, which makes this concept easy to grok.

### PR DESCRIPTION
**Pull request by leerob**
Closes https://github.com/vercel/next.js/issues/20369#event-4136491600.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| buildDuration | 9s | 9.2s | ⚠️ +239ms |
| nodeModulesSize | 82.6 MB | 82.6 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.172 | 2.189 | ⚠️ +0.02 |
| / avg req/sec | 1151 | 1141.83 | ⚠️ -9.17 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.125 | 1.174 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 2221.97 | 2129.72 | ⚠️ -92.25 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| 677f882d2ed8..5e70.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-1f21dcf..0011.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| index.html gzip | 612 B | 612 B | ✓ |
| link.html gzip | 619 B | 619 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| buildDuration | 11s | 10.9s | -62ms |
| nodeModulesSize | 82.6 MB | 82.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| 677f882d2ed8..5e70.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-1f21dcf..0011.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-2cf0e..a0a0.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.31 kB | 8.31 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: a4ab76a00bee2a5732129a7c0f6f9163fe44b054

File: docs/deployment.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ For example, the [hybrid pages](/docs/basic-features/pages.md) approach is fully
 - **Automatic HTTPS:** HTTPS is enabled by default (including custom domains) and doesn't require extra configuration. We auto-renew SSL certificates.
 - **More:** [Read our documentation](https://vercel.com/docs) to learn more about the Vercel platform.
 
+### Automatic Updates
```
Comment (lfades) [inline docs/deployment.md:53]:
```suggestion
## Automatic Updates
```

Comment (leerob) [reply docs/deployment.md]: Thanks!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| buildDuration | 8.4s | 8.4s | ⚠️ +33ms |
| nodeModulesSize | 82.6 MB | 82.6 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.664 | 1.679 | ⚠️ +0.02 |
| / avg req/sec | 1502.66 | 1488.83 | ⚠️ -13.83 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.061 | 1.055 | -0.01 |
| /error-in-render avg req/sec | 2355.63 | 2370.21 | +14.58 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| 677f882d2ed8..5e70.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-72193d1..1356.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-85785..a9f3.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.05 kB | 8.05 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 618 B | 618 B | ✓ |
| withRouter.html gzip | 607 B | 607 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| buildDuration | 9.8s | 9.8s | ⚠️ +35ms |
| nodeModulesSize | 82.6 MB | 82.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| 677f882d2ed8..5e70.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-72193d1..1356.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-95c2..e870.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| polyfills-d3..23f6.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _app-0d19cb6..5497.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-85785..a9f3.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-42456f..0c06.js gzip | 887 B | 887 B | ✓ |
| index-8081ce..e44f.js gzip | 227 B | 227 B | ✓ |
| link-0ab9f83..fa00.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..c3d8.js gzip | 303 B | 303 B | ✓ |
| withRouter-0..a68e.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.05 kB | 8.05 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | leerob/next.js deployment-with-new-versions | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: 675683d04a40e37fbc8e0e520fd22e238e808e03

**Review by timneutkens — APPROVED** [review]

**Pull request merged**